src/v/http/client.cc

```diff
@@ -247,10 +247,10 @@ ss::future<> client::response_stream::shutdown() { return _client->stop(); }
 /// Return failed future if ec is set, otherwise return future in ready state
 static ss::future<iobuf>
 fail_on_error(prefix_logger& ctxlog, const boost::beast::error_code& ec) {
-    if (!ec) {
+    if (!ec.failed()) {
```

Comment:
nit: why this change is needed?

Reply:
not needed, but I think that's more readable